iceoryx2-ffi/cxx/tests/src/service_publish_subscribe_tests.cpp

```diff
@@ -314,4 +320,44 @@ TYPED_TEST(ServicePublishSubscribeTest, publisher_applies_unable_to_deliver_stra
     ASSERT_THAT(sut_pub_1.unable_to_deliver_strategy(), Eq(UnableToDeliverStrategy::Block));
     ASSERT_THAT(sut_pub_2.unable_to_deliver_strategy(), Eq(UnableToDeliverStrategy::DiscardSample));
 }
+
+TYPED_TEST(ServicePublishSubscribeTest, send_receive_with_user_header_works) {
+    constexpr ServiceType SERVICE_TYPE = TestFixture::TYPE;
+
+    const auto* name_value = "I am floating through the galaxy of my brain. Oh the colors!";
+    const auto service_name = ServiceName::create(name_value).expect("");
+
+    auto node = NodeBuilder().create<SERVICE_TYPE>().expect("");
+    auto service_pub = node.service_builder(service_name)
+                           .template publish_subscribe<uint64_t>()
+                           .template user_header<TestHeader>()
```

Comment:
```suggestion
                           .publish_subscribe<uint64_t>()
                           .user_header<TestHeader>()
```

Reply:
does not compile.

Reply:
I'm disappointed you gave up so easily ... just write a new C++ compiler ;)